sdk/go/things.go

```diff
@@ -192,17 +176,13 @@ func (sdk mfSDK) Thing(id, token string) (Thing, error) {
 	}
 
 	if resp.StatusCode != http.StatusOK {
-		switch resp.StatusCode {
-		case http.StatusForbidden:
-			return Thing{}, ErrUnauthorized
-		case http.StatusNotFound:
-			return Thing{}, ErrNotFound
-		default:
-			return Thing{}, ErrFetchFailed
+		if err := encodeError(resp.StatusCode); err != nil {
+			return Thing{}, err
 		}
+		return Thing{}, ErrFetchFailed
 	}
 
-	var t Thing
+	t := Thing{}
```

Comment:
I think its cleaner if you leave it as before `var t Thing` instead of a short declaration. You don't need to declare anything here, zero values are just fine.